docs/pages/deploy-a-cluster/high-availability.mdx

```diff
@@ -0,0 +1,369 @@
+  `teleport` instances can access. This requires permissions for `teleport`
+  instances to manage records within the key-value store.
+- An **object storage service** for session recordings uploaded by the Auth
+  Service. This requires permissions for `teleport` instances to manage objects
+  within the storage service.
+- An **automated TLS credential provisioner** that obtains TLS credentials from
+  Let's Encrypt, renews the credentials, and provisions `teleport` instances
+  with them.
+- A **DNS service** you can use to create records for the Teleport Proxy
+  Service. The TLS credential provisioner will also use this to create records
+  to demonstrate control over your domain name and fetch credentials from Let's
+  Encrypt.
+
+{/* TODO: include a diagram--do this after filling in the rest of the details*/}
+
+## Layer Four load balancer
+
+The load balancer forwards traffic from users and services to an available
+Teleport instance. This must not terminate TLS, and must transparently forward
+the TCP traffic it receives. In other words, this must be a Layer Four load
+balancer, not a Layer Seven (e.g., HTTP) load balancer. 
+
+Configure the load balancer to forward traffic from the following ports on the
+load balancer to the corresponding port on an available Teleport instance:
+
+| Port | Description |
+| - | - |
+| `3021` | Port used by Teleport Proxy Service instances to dial agents in [Proxy Peering mode](../management/operations/proxy-peering.mdx). |
+| `3022` | SSH port. This is Teleport's equivalent of port `22` for SSH. |
+| `3023` | SSH port for clients connect to. The Proxy Service will forward this connection to port `3022` on the destination Node. |
+| `3024` | SSH port used to create reverse SSH tunnels from behind-firewall environments. |
+| `3025` | TLS port used by the Auth Service to serve its gRPC API to Teleport services  in a cluster. |
+| `443` | HTTPS connections to authenticate `tsh` users into the cluster. The same connection is used to serve a Web UI. |
+| `3026` | HTTPS Kubernetes proxy |
+| `3027` | Kubernetes Service |
+| `3028` | Desktop Service |
+| `3036` | MySQL port |
+| `5432` | Postgres port |
+
+## Key-value store
+
+The Teleport Auth Service stores cluster state (such as dynamic configuration
+resources) and audit events as key/value pairs. In high-availability
+deployments, you must configure the Auth Service to manage this data in a
+key-value store that runs outside of your cluster of Teleport instances.
+
+The Auth Service supports the following backends for cluster state and audit
+events:
+
+- Amazon DynamoDB
+- Google Cloud Firestore
+- etcd
+
+For Amazon DynamoDB and Google Cloud Firestore, your Teleport configuration (which
+we will describe in more detail in the [Configuration](#configuration) section)
+names a table or collection where Teleport stores cluster state and audit
+events. For etcd, Teleport uses namespaces within item keys to identify audit
```

Comment:
etcd is not supported for audit event storage, we have no on-prem solution for audit event storage at this time